include/aos/common/uuid.hpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_UUID_HPP_
+#define AOS_UUID_HPP_
+
+#include "aos/common/tools/string.hpp"
+
+namespace aos {
+namespace uuid {
+
+/**
+ *  UUID length
+ */
+constexpr auto cUUIDLen = AOS_CONFIG_UUID_LEN;
+
+/**
+ *  Length of UUID string representation
+ */
+constexpr auto cUUIDStrLen = AOS_CONFIG_UUID_STR_LEN;
+
+/**
+ * A UUID is a 128 bit (16 byte) Universal Unique IDentifier as defined in RFC 4122.
+ */
+using UUID = StaticArray<uint8_t, cUUIDLen>;
+
+/**
+ * Creates unique UUID.
+ *
+ * @param dst result uuid.
+ * @return Error.
+ */
+inline Error CreateUUID(UUID& dst)
```

Comment:
We definitely could return it through stack: `RetWithError`.
Also, don't think all these functions will be inlined.